app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -147,4 +173,47 @@ class TabsTrayFragment : AppCompatDialogFragment(), TabsTrayInteractor {
             isUserInputEnabled = false
         }
     }
+
+    private fun setupMenu(view: View) {
+        view.tab_tray_overflow.setOnClickListener { anchor ->
+            val tabTrayItemMenu =
+                TabsTrayMenu(
+                    context = view.context,
+                    browserStore = requireComponents.core.store,
+                    tabLayout = tab_layout
+                ) {
+                    when (it) {
+                        is TabsTrayMenu.Item.ShareAllTabs ->
+                            navigationInteractor.onShareTabsOfTypeClicked(isPrivateModeSelected)
+                        is TabsTrayMenu.Item.OpenTabSettings ->
+                            navigationInteractor.onTabSettingsClicked()
+                        is TabsTrayMenu.Item.CloseAllTabs ->
+                            navigationInteractor.onCloseAllTabsClicked(isPrivateModeSelected)
+                        is TabsTrayMenu.Item.OpenRecentlyClosed ->
+                            navigationInteractor.onOpenRecentlyClosedClicked()
+                        is TabsTrayMenu.Item.SelectTabs ->
+                        { /* TODO implement when mulitiselect call is available */ }
+                    }
+                }
+
+            requireComponents.analytics.metrics.track(Event.TabsTrayMenuOpened)
+            val menu = tabTrayItemMenu.menuBuilder.build(view.context)
+            menu.show(anchor).also { popupMenu ->
+                (popupMenu.contentView as? CardView)?.setCardBackgroundColor(
+                    ContextCompat.getColor(
+                        view.context,
+                        R.color.foundation_normal_theme
+                    )
+                )
+            }
+        }
+    }
+    private val homeViewModel: HomeScreenViewModel by activityViewModels()
+
+    private fun dismissTabTrayAndNavigateHome(sessionId: String) {
+        homeViewModel.sessionToDelete = sessionId
+        val directions = NavGraphDirections.actionGlobalHome()
+        findNavController().navigate(directions)
+        dismissAllowingStateLoss()
```

Comment:
Sorry, I missed this in the last pass of the review.

Are we doing the same thing as `navigateToBrowser`? If so, then maybe:
```suggestion
        navigateToBrowser()
```

Reply:
I tried replacing with `navigateToBrowser` but it doesn't navigate to browser home.